ucp/_libs/src/ucp_py_ucp_fxns.c

```diff
@@ -608,11 +631,43 @@ void *ucp_py_get_ep(char *ip, int listener_port)
                               send_handle);
     if (UCS_PTR_IS_ERR(request)) {
         fprintf(stderr, "unable to send UCX data message\n");
+	if (EP_CONN_FAILED == connection_status) {
+            ucp_request_cancel(ucp_py_ctx_head->ucp_worker, &request);
+            request_init(request);
+            ucp_request_free(request);
+            printf("connection failed. Retrying in %d seconds\n",
+                        ucp_py_ctx_head->connection_retry_period);
+            connection_status = EP_CONN_INIT;
+            sleep(ucp_py_ctx_head->connection_retry_period);
+            if (num_retries < EP_CONN_MAX_RETRIES) {
```

Comment:
It's common to provide a timeout in seconds.  For example I might expect something like the following:

```python
connection = await connect('127.0.0.1:8000', timeout=10)
```

Reply:
This is probably cleaner the env variable I'm using now.

Reply:
Do we want to include `max_retries` as well?

Reply:
In applications that I've seen it's common to specify only a timeout